app/src/main/java/org/thoughtcrime/securesms/ui/theme/ThemeFromPreferences.kt

```diff
@@ -17,38 +17,28 @@ import org.session.libsession.utilities.TextSecurePreferences.Companion.YELLOW_A
  * Some behaviour is hardcoded to cater for legacy usage of people with themes already set
  * But future themes will be picked and set directly from the "Appearance" screen
  */
-@Composable
-fun TextSecurePreferences.getComposeTheme(): ThemeColors {
+fun TextSecurePreferences.getColorSet(): ThemeColorSet {
     val selectedTheme = getThemeStyle()
 
     // get the chosen primary color from the preferences
     val selectedPrimary = primaryColor()
 
-    // create a theme set with the appropriate primary
-    val colorSet = when(selectedTheme){
-        TextSecurePreferences.OCEAN_DARK,
-        TextSecurePreferences.OCEAN_LIGHT -> ThemeColorSet(
-            light = OceanLight(selectedPrimary),
-            dark = OceanDark(selectedPrimary)
-        )
-
-        else -> ThemeColorSet(
-            light = ClassicLight(selectedPrimary),
-            dark = ClassicDark(selectedPrimary)
-        )
-    }
+    val createLight = if ("ocean" in selectedTheme) ::OceanLight else ::ClassicLight
+    val createDark = if ("ocean" in selectedTheme) ::OceanDark else ::ClassicDark
 
-    // deliver the right set from the light/dark mode chosen
-    val theme = when{
-        getFollowSystemSettings() -> if(isSystemInDarkTheme()) colorSet.dark else colorSet.light
+    val followSystemSettings = getFollowSystemSettings()
 
-        selectedTheme == TextSecurePreferences.CLASSIC_LIGHT ||
-                selectedTheme == TextSecurePreferences.OCEAN_LIGHT -> colorSet.light
+    return if (followSystemSettings) ThemeColorSet(
+        colorsWhenSystemInLight = createLight(selectedPrimary),
+        colorsWhenSystemInDark = createDark(selectedPrimary)
+    ) else {
+        val both = if ("light" in selectedTheme) createLight(selectedPrimary) else createDark(selectedPrimary)
```

Comment:
`both` is a confusing name here. It would be better as something like `selectedColors`, or something like that.

Reply:
yeah, gone